base/Pose.hpp

```diff
@@ -37,35 +39,89 @@ namespace base
         return res;
     }
 
+    /*
+     * Decomposes the orientation in euler angles so that this can be
+     * obtained by applying the following rotations in order:
+     *
+     *  rotation of a2 around x-axis
+     *  rotation of a1 around y-axis
+     *  rotation of a0 around z-axis
+     *
+     * assuming angles in range of: a0:(-pi,pi), a1:(-pi/2,pi/2), a2:(-pi/2,pi/2)
+     *
+     */
+    static base::Vector3d getEuler(const base::AngleAxisd &orientation){
+        const Eigen::Matrix3d m = orientation.toRotationMatrix();
+        double x = base::Vector2d(m.coeff(2,2) , m.coeff(2,1)).norm();
+        base::Vector3d res(0,::atan2(-m.coeff(2,0), x),0);
+        if (x > Eigen::NumTraits<double>::dummy_precision()){
+            res[0] = ::atan2(m.coeff(1,0), m.coeff(0,0));
+            res[2] = ::atan2(m.coeff(2,1), m.coeff(2,2));
+        }else{
+            res[0] = 0;
+            res[2] = (m.coeff(2,0)>0?1:-1)* ::atan2(-m.coeff(0,1), m.coeff(1,1));
+        }
+        return res;
+    }
+
     static double getYaw(const base::Orientation& orientation)
     {
         return base::getEuler(orientation)[0];
     }
 
+    static double getYaw(const base::AngleAxisd& orientation)
+    {
+        return base::getEuler(orientation)[0];
+    }
+
     static double getPitch(const base::Orientation& orientation)
     {
         return base::getEuler(orientation)[1];
     }
 
+    static double getPitch(const base::AngleAxisd& orientation)
+    {
+        return base::getEuler(orientation)[1];
+    }
+
     static double getRoll(const base::Orientation& orientation)
     {
         return base::getEuler(orientation)[2];
     }
 
+    static double getRoll(const base::AngleAxisd& orientation)
+    {
+        return base::getEuler(orientation)[2];
+    }
 
     static inline base::Orientation removeYaw(const base::Orientation& orientation)
     {
-	return Eigen::AngleAxisd( -getYaw(orientation), Eigen::Vector3d::UnitZ()) * orientation;
+	    return Eigen::AngleAxisd( -getYaw(orientation), Eigen::Vector3d::UnitZ()) * orientation;
```

Comment:
You indent with tabs and have your editor setup with tab=4 spaces. Fucks up indentation on e.g. github and anyone that has tab == 8 spaces (which is the default for most editors). Please stick to 4-space indent, no tabs.